cmd/integration-test/pkg/clientconfig/clientconfig.go

```diff
@@ -144,24 +145,38 @@ func SignHTTPRequestWithEmail(ctx context.Context, client *client.Client, req *h
 	return msg.Sign(email, newKey)
 }
 
+var talosAPIKeyMutex sync.Mutex
+
 // TalosAPIKeyPrepare prepares a public key to be used with tests interacting via Talos API client using the default test email.
 func TalosAPIKeyPrepare(ctx context.Context, client *client.Client, contextName string) error {
 	return TalosAPIKeyPrepareWithEmail(ctx, client, contextName, defaultEmail)
 }
 
 // TalosAPIKeyPrepareWithEmail prepares a public key to be used with tests interacting via Talos API client using the given email.
 func TalosAPIKeyPrepareWithEmail(ctx context.Context, client *client.Client, contextName, email string) error {
-	newKey, err := pgp.GenerateKey("", "", email, 4*time.Hour)
+	talosAPIKeyMutex.Lock()
```

Comment:
need this with the new api-signature library?

Reply:
yeah, that's another race, in the unrelated flow.
This function is called in the tests which can run in parallel.